include/dd/Package.hpp

```diff
@@ -1526,6 +1527,43 @@ namespace dd {
             return {CTEntry::val(c.r), CTEntry::val(c.i)};
         }
 
+    public:
+        fp expectationValue(const mEdge& x, const vEdge& y) {
+            /**
+            Calculates the expectation value of an operator x with respect to a quantum state y given their corresponding decision diagrams.
+            @param x a matrix DD representing the operator
+            @param y a vector DD representing the quantum state
+            @return a floating point value representing the expectation value of the operator with respect to the quantum state
+            @throw an exception message is thrown if the edges are not on the same level or if the expectation value is non-real.
+            @note This function calls the multiply() function to apply the operator to the quantum state, then calls innerProduct()
+                  to calculate the overlap between the original state and the applied state i.e. <Psi| Psi'> = <Psi| (Op|Psi>).
+                  It also calls the garbageCollect() function to free up any unused memory.
+            **/
+
+            try {
+                if (x.p->v != y.p->v) {
+                    throw("Edges are not on same level. Expectation Value cannot be calculated.");
+                }
+            } catch (const char* message) {
+                std::cout << message;
+            }
```

Comment:
No need to wrap this in a `try ... catch` block. Immediately catching the exception and just printing a message defeats the whole purpose of throwing an exception.

Also, it's common practice to throw dedicated exceptions from the available exception classes (see, e.g., https://en.cppreference.com/w/cpp/error/exception)

This should probably be something like
```suggestion
            if (x.p->v != y.p->v) {
                throw std::invalid_argument("Observable and state must act on the same number of qubits to compute the expectation value.");
            }
```